lib/segment/src/segment/segment_ops.rs

```diff
@@ -72,6 +72,7 @@ impl Segment {
     /// # Warning
     ///
     /// Available for appendable segments only.
+    #[allow(clippy::needless_pass_by_ref_mut)] // ensure single access to AtomicRefCell vector_index
```

Comment:
the important change

Reply:
Nice. I was looking for cases like this, thank you for carefully reviewing.